### DISCUSSION TITLE
Could not parse expression

### DISCUSSION THREAD

**Discussion by RustamSultansoy** [Blazebit/blaze-persistence#1696] [Q&A] [answered]
Hello. I'm using blazebit with postgresql. Now i've started to use postgres arrays with Vlad Mihalcea's ListArrayType. And when i try to query something using where expression like: c.whereExpression(16=ANY(some_array::INT[])) i get and error Could not parse expression '16=ANY(some_array::INT[])', line 1:45 no viable alternative at input '16=ANY(some_array:' 
is there a good way to get list by value in array?

**Comment by beikov** [comment, marked as answer]
Hi. Usually, this is modeled by registering a custom function i.e. `any_int` and using it in your expressions like `16=any_int(someArray)`.
You can read more about custom functions and how to register them in the [documentation](https://persistence.blazebit.com/documentation/1.6/core/manual/en_US/#custom-jpql-functions). It will roughly look like this:

```java
public class AnyIntFunction implements JpqlFunction {

    @Override
    public boolean hasArguments() {
        return true;                            
    }

    @Override
    public boolean hasParenthesesIfNoArguments() {
        return true;                            
    }

    @Override
    public Class<?> getReturnType(Class<?> firstArgumentType) {
        return firstArgumentType;               
    }

    @Override
    public void render(FunctionRenderContext context) { 
        context.addChunk("any(");
        context.addArgument(0);                
        context.addChunk("::int[])");
    }

}
```

**Reply by RustamSultansoy** [reply]
thanks a lot
